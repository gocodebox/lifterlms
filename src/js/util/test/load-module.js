import loadModule from '../load-module';

const mockModule = {
	test: 1,
	test2: 'abc',
	test3: [ 4, 5, 6 ],
	test4: () => {},
};

describe( 'loadModule', () => {
	beforeEach( () => {
		global.window = {};
	} );

	it( 'should load the module in the global llms object when llms is yet to be defined', () => {
		loadModule( 'mockModule', mockModule );

		expect( window.llms.mockModule ).toEqual( mockModule );
		expect( window.llms.mockModule ).not.toBe( mockModule );
	} );

	it( 'should load the module in the global llms object when llms is already defined', () => {
		global.window.llms = { some: 'stuff', already: 9320 };

		loadModule( 'mockModule', mockModule );

		expect( window.llms.mockModule ).toEqual( mockModule );
		expect( window.llms.mockModule ).not.toBe( mockModule );

		expect( window.llms.some ).toEqual( 'stuff' );
		expect( window.llms.already ).toEqual( 9320 );
	} );

	it( 'should extend an object that already exists at the specified key when extend is true', () => {
		global.window.llms = { mockModule: { a: 1, b: 2 } };

		loadModule( 'mockModule', mockModule, true );

		expect( window.llms.mockModule ).toMatchSnapshot();
	} );

	it( 'should overwrite an object that already exists at the specified key when extend is false', () => {
		global.window.llms = { mockModule: { a: 1, b: 2 } };

		loadModule( 'mockModule', mockModule );

		expect( window.llms.mockModule ).toEqual( mockModule );
		expect( window.llms.mockModule ).not.toBe( mockModule );
	} );
} );

