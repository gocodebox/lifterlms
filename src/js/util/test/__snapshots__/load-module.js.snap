// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadModule should extend an object that already exists at the specified key when extend is true 1`] = `
Object {
  "a": 1,
  "b": 2,
  "test": 1,
  "test2": "abc",
  "test3": Array [
    4,
    5,
    6,
  ],
  "test4": [Function],
}
`;
